VTIL-Tests/dummy.cpp

```diff
@@ -1,492 +1,555 @@
+
+    vtil::optimizer::stack_pinning_pass{}(block->owner);
+
+    vtil::logger::log(":: After:\n");
+    vtil::debug::dump(block->owner);
+
+    CHECK(block->size() == 6);
+    CHECK(block->sp_offset == 0);
+
+}
+
+
+DOCTEST_TEST_CASE("Optimization istack_ref_substitution_pass")
+{
+    vtil::logger::log("\n\n>> %s \n", __FUNCTION__);
+    auto block = vtil::basic_block::begin(0x1337);
+    vtil::register_desc reg_eax(vtil::register_physical, registers::ax, vtil::arch::bit_count, 0);
+
+    // mov eax,esp
+    block->mov(reg_eax, vtil::REG_SP);
+    // sub eax, 4
+    block->sub(reg_eax, vtil::arch::size);
+    // mov [eax+0], 1
+    block->str(reg_eax, 0, (uintptr_t) 1);
+    block->vexit(0ull); // marks the end of a basic_block
+
+    vtil::logger::log(":: Before:\n");
+    vtil::debug::dump(block->owner);
+
+    vtil::optimizer::istack_ref_substitution_pass{}(block->owner);
+
+    vtil::logger::log(":: After:\n");
+    vtil::debug::dump(block->owner);
+
+    auto ins = (*block) [2];
+
+    // mov [eax-4], 1
+    CHECK(ins.base == &vtil::ins::str);
+    CHECK(ins.operands.size() == 3);
+    CHECK(ins.operands[0].reg().to_string() == "$sp");
+    CHECK(ins.operands[1].imm().ival == -(intptr_t)(vtil::arch::size));
+    CHECK(ins.operands[2].imm().ival == 0x1);
+
+}
+
+
+DOCTEST_TEST_CASE("Optimization stack_propagation_pass")
+{
+    vtil::logger::log("\n\n>> %s \n", __FUNCTION__);
+
+    auto block = vtil::basic_block::begin(0x1337);
+    vtil::register_desc reg_eax(vtil::register_physical, registers::ax, vtil::arch::bit_count, 0);
+
+
+    // mov [esp+0], 0x1234
+    block->str(vtil::REG_SP, 0, (uintptr_t)0x1234);
+    // mov eax, [esp+0]
+    block->ldd(reg_eax, vtil::REG_SP, 0);
+    block->vexit(0ull); // marks the end of a basic_block
+
+    vtil::logger::log(":: Before:\n");
+    vtil::debug::dump(block->owner);
+
+    vtil::optimizer::stack_propagation_pass{}(block->owner);
+
+    vtil::logger::log(":: After:\n");
+    vtil::debug::dump(block->owner);
+
+    auto ins = (*block)[1];
+
+    // mov eax, 0x1234
+    CHECK(ins.base == &vtil::ins::mov);
+    CHECK(ins.operands.size() == 2);
+    CHECK(ins.operands[0].reg().local_id == registers::ax);
+    CHECK(ins.operands[1].imm().ival == 0x1234);
+}
+
+
+DOCTEST_TEST_CASE("Optimization dead_code_elimination_pass")
+{
+    vtil::logger::log("\n\n>> %s \n", __FUNCTION__);
+
+    auto block = vtil::basic_block::begin(0x1337);
+    vtil::register_desc reg_eax(vtil::register_physical, registers::ax, vtil::arch::bit_count, 0);
+
+
+    block->mov(reg_eax, (uintptr_t) 1);
+    block->mov(reg_eax, (uintptr_t) 2);
+    block->vexit(0ull); // marks the end of a basic_block
+
+    vtil::logger::log(":: Before:\n");
+    vtil::debug::dump(block->owner);
+
+    vtil::optimizer::dead_code_elimination_pass{}(block->owner);
+
+    vtil::logger::log(":: After:\n");
+    vtil::debug::dump(block->owner);
+
+    CHECK(block->size() == 2);
+}
+
+
+DOCTEST_TEST_CASE("Optimization mov_propagation_pass")
```

Comment:
Could you fix these newlines so it’s possible to see what you changed? Now the whole file looks deleted.

Reply:
maybe problem about the commit ca67b74dbd7cd64a069eb7a00e2a3215d11007ad. 
I edited this file online to resolve the confilct. sames github cant trans \r\n 😂

Reply:
Alright, just leave it then. I’ll check on the PC